pkg/envoy/envoy.go

```diff
@@ -127,15 +140,25 @@ func GetEnvoyVersion() string {
 }
 
 // StartEnvoy starts an Envoy proxy instance.
-func StartEnvoy(adminPort uint32, stateDir, logPath string, baseID uint64) *Envoy {
+func StartEnvoy(stateDir, logPath string, baseID uint64) *Envoy {
 	bootstrapPath := filepath.Join(stateDir, "bootstrap.pb")
-	adminAddress := "127.0.0.1:" + strconv.FormatUint(uint64(adminPort), 10)
 	xdsPath := getXDSPath(stateDir)
 
+	// Have to use a fake IP address:port even when we Dial to a Unix domain socket.
+	// The address:port will be visible to Envoy as ':authority', but it's value is
+	// not meaningful.
+	// Not using the normal localhost address to make it obvious that we are not
+	// connecting to Envoy's admin interface via the IP stack.
+	adminAddress := "128.1.2.3:42"
```

Comment:
I'd recommend using a reserved IP address instead of a valid one, e.g. `192.0.2.34:56`.

Reply:
Good point, changed!